elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -603,53 +607,58 @@ export class PeerPool extends EventEmitter {
 		);
 
 		// Cannot predict which netgroups will be protected
-		const filteredPeersByNetgroup = this._peerPoolConfig.netgroupProtectionRatio
+		const protectedPeersByNetgroup = this._peerPoolConfig
+			.netgroupProtectionRatio
 			? filterPeersByCategory(peers, {
 					category: PROTECTION_CATEGORY.NET_GROUP,
 					percentage: this._peerPoolConfig.netgroupProtectionRatio,
 					asc: true,
 			  })
-			: peers;
-		if (filteredPeersByNetgroup.length <= 1) {
-			return filteredPeersByNetgroup;
-		}
+			: [];
 
 		// Cannot manipulate without physically moving nodes closer to the target.
-		const filteredPeersByLatency = this._peerPoolConfig.latencyProtectionRatio
+		const protectedPeersByLatency = this._peerPoolConfig.latencyProtectionRatio
 			? filterPeersByCategory(peers, {
 					category: PROTECTION_CATEGORY.LATENCY,
 					percentage: this._peerPoolConfig.latencyProtectionRatio,
-					asc: true,
+					asc: false,
 			  })
-			: filteredPeersByNetgroup;
-		if (filteredPeersByLatency.length <= 1) {
-			return filteredPeersByLatency;
-		}
+			: [];
 
 		// Cannot manipulate this metric without performing useful work.
-		const filteredPeersByResponseRate = this._peerPoolConfig
+		const protectedPeersByResponseRate = this._peerPoolConfig
 			.productivityProtectionRatio
-			? filterPeersByCategory(filteredPeersByLatency, {
+			? filterPeersByCategory(peers, {
 					category: PROTECTION_CATEGORY.RESPONSE_RATE,
 					percentage: this._peerPoolConfig.productivityProtectionRatio,
-					asc: false,
+					asc: true,
 			  })
-			: filteredPeersByLatency;
-		if (filteredPeersByResponseRate.length <= 1) {
-			return filteredPeersByResponseRate;
-		}
+			: [];
+
+		const uniqueProtectedPeers = [
+			...new Set([
+				...protectedPeersByNetgroup,
+				...protectedPeersByLatency,
+				...protectedPeersByResponseRate,
+			]),
+		];
+		const unprotectedPeers = peers.filter(
+			peer => !uniqueProtectedPeers.includes(peer),
+		);
 
-		// Protect remaining half of peers by longevity, precludes attacks that start later.
-		const filteredPeersByConnectTime = this._peerPoolConfig
+		// Protect *the remaining half* of peers by longevity, precludes attacks that start later.
+		const protectedPeersByConnectTime = this._peerPoolConfig
 			.longevityProtectionRatio
-			? filterPeersByCategory(filteredPeersByResponseRate, {
+			? filterPeersByCategory(unprotectedPeers, {
 					category: PROTECTION_CATEGORY.CONNECT_TIME,
 					percentage: this._peerPoolConfig.longevityProtectionRatio,
-					asc: true,
+					asc: false,
```

Comment:
Doesn't this mean that those with the shortest connected time are more likely to be protected? We need to protect the oldest ones with the longest connected time.

Reply:
No, it means those who connected at the smallest `Date` number e.g. longest connected time.